src/main/java/com/opentable/kafka/builders/KafkaConsumerBuilderFactoryBean.java

```diff
@@ -0,0 +1,35 @@
+package com.opentable.kafka.builders;
+
+import java.util.Optional;
+import java.util.Properties;
+
+import javax.inject.Inject;
+
+import com.codahale.metrics.MetricRegistry;
+
+import org.springframework.core.env.ConfigurableEnvironment;
+
+import com.opentable.service.AppInfo;
+import com.opentable.service.ServiceInfo;
+
+public class KafkaConsumerBuilderFactoryBean<K,V> extends KafkaBaseBuilderFactoryBean {
+
+    @Inject
+    public KafkaConsumerBuilderFactoryBean(Optional<ServiceInfo> serviceInfo, AppInfo appInfo, ConfigurableEnvironment env, Optional<MetricRegistry> metricRegistry) {
+        super(serviceInfo, appInfo, env, metricRegistry);
+    }
+
```

Comment:
A separate builderfactory bean for each so that generics flow correctly. If you can reunite them, fine, but I'm not convinced that's possibble